### PR TITLE
gh-100734: Add 'Notable change in 3.11.x' to `whatsnew/3.11.rst`

### PR DESCRIPTION
**Pull request by hugovk**
These were added to the 3.11 branch, but are missing from 3.12 and `main`:

* ✅ https://docs.python.org/3.11/whatsnew/3.11.html#notable-changes-in-3-11-4
* ❌ https://docs.python.org/3.12/whatsnew/3.11.html#notable-changes-in-3-11-4
* ❌ https://docs.python.org/dev/whatsnew/3.11.html#notable-changes-in-3-11-4

----
📚 Documentation preview 📚: https://cpython-previews--114657.org.readthedocs.build/

* Issue: gh-100734

### CODE REVIEW

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -2701,4 +2701,29 @@ Removed
...
+
+
+Notable changes in 3.11.5
+=========================
+
+OpenSSL
+-------
+
+* Windows builds and macOS installers from python.org now use OpenSSL 3.0.
+
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst:2728]:
```suggestion


```

File: Doc/whatsnew/3.11.rst [hunk 2]
```diff
@@ -2701,4 +2701,29 @@ Removed
   (Contributed by Inada Naoki in :issue:`44029`.)
 
 
+Notable changes in 3.11.4
+=========================
+
+tarfile
+-------
+
+* The extraction methods in :mod:`tarfile`, and :func:`shutil.unpack_archive`,
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst:2710]: Perhaps an item list is not needed for a single item.

**Review by serhiy-storchaka — APPROVED** [review]

File: Doc/whatsnew/3.11.rst (hunk 2 above)
Comment (hugovk) [reply Doc/whatsnew/3.11.rst:2710]: This came up in the 3.12 What's New, and we decided to keep it so the indent is consistent with the others, and it looks better as you scroll down.

**Comment by hugovk** [comment]
I happened to find gh-100734, this is the same sort of thing so let's link them.

**Pull request merged**